### PR TITLE
chore(data-service): remove external usage of DataService internals COMPASS-5057

### PR DESCRIPTION
**Pull request by rose-m**
## Description
This PR removes usages of `DataService` internal members across Compass. There only remains one usage of `DataService.model` in `compass-sidebar/store.js` which can only be replaced once the `ConnectionStore` (https://jira.mongodb.org/browse/COMPASS-5010) is implemented.

**The changes in this PR change the API of the package.** This also requires the VSCode extension to be adjusted once a new version is published and VSCode extension wants to upgrade (cc @alenakhineika).

**Important:** This PR also changes `hadron-ipc` to not just immediately fail when required in a non-electron environment (for tests).

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context
- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [x] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/hadron-ipc/index.js [hunk 1]
```diff
@@ -2,8 +2,19 @@
 
 const isRenderer = require('is-electron-renderer');
 
-if (isRenderer) {
-  module.exports = require('./lib/renderer');
-} else {
-  module.exports = require('./lib/main');
+const isElectron = typeof require('electron') !== 'string';
```
Comment (addaleax) [inline packages/hadron-ipc/index.js:5]: This assumes that `electron` is always present as a peer dependency, which it might not be.

Comment (addaleax) [reply packages/hadron-ipc/index.js:5]:
```suggestion
let isElectron = false;
try {
  isElectron = typeof require('electron') !== 'string';
} catch { /* not electron */ }
```

File: packages/data-service/src/connection-title.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import ConnectionString from 'mongodb-connection-string-url';
+import { ConnectionOptions } from './connection-options';
+
+export function getConnectionTitle(options: ConnectionOptions): string {
+  if (options.favorite?.name) {
+    return options.favorite.name;
+  }
+
+  const url = new ConnectionString(options.connectionString);
+  if (url.isSRV && url.hosts.length === 1) {
```
Comment (addaleax) [inline packages/data-service/src/connection-title.ts:10]: `.isSRV` implies `.hosts.length === 1` already, fwiw.

**Review by addaleax — APPROVED** [review]

File: packages/compass-export-to-language/src/stores/store.js [hunk 3]
```diff
@@ -31,7 +31,9 @@ export const setCopyToClipboardFn = (store, fn) => {
  * @param {Object} dataService - The data provider.
  */
 export const setDataProvider = (store, error, dataService) => {
-  store.dispatch(uriChanged(dataService.model.driverUrl));
+  // TODO: verify - do we want the _driver_ URL we modified (directConnection, etc.) -> getMongoClientConnectionOptions().url
+  // or do we pick the user's connection string -> getConnectionOptions().connectionString
+  store.dispatch(uriChanged(dataService.getConnectionOptions().connectionString));
```
[the remark this answers is not in the document]
Comment (mcasimir) [reply packages/compass-export-to-language/src/stores/store.js:36]: >  it’s probably least confusing if we use the connection string that we received from the user.

That should be what Michael did here, looks like `getConnectionOptions` returns the options passed to the DataService constructor.

**Comment by mcasimir** [comment]
Replaced by #2475

**Pull request closed without merging**